starknet_py/net/full_node_client.py

```diff
@@ -428,14 +443,16 @@ def get_block_identifier(
 
 def _get_call_payload(tx: Union[InvokeFunction, Call]) -> dict:
     if isinstance(tx, InvokeFunction):
+        invoke = cast(InvokeFunction, tx)
         return {
-            "contract_address": convert_to_felt(tx.contract_address),
-            "entry_point_selector": convert_to_felt(tx.entry_point_selector),
-            "calldata": [convert_to_felt(i) for i in tx.calldata],
+            "contract_address": convert_to_felt(invoke.contract_address),
+            "entry_point_selector": convert_to_felt(invoke.entry_point_selector),
+            "calldata": [convert_to_felt(i) for i in invoke.calldata],
         }
 
+    call = cast(Call, tx)
     return {
-        "contract_address": convert_to_felt(tx.to_addr),
-        "entry_point_selector": convert_to_felt(tx.selector),
-        "calldata": [convert_to_felt(i) for i in tx.calldata],
+        "contract_address": convert_to_felt(call.to_addr),
+        "entry_point_selector": convert_to_felt(call.selector),
+        "calldata": [convert_to_felt(i) for i in call.calldata],
     }
```

Comment:
These cast seem quite redundant to me. Isn't it possible to make pyright somehow work with isinstance calls?

Reply:
This works for some reason:
```
    if isinstance(tx, InvokeFunction):
        invoke = cast(InvokeFunction, tx)
        return {
            "contract_address": convert_to_felt(invoke.contract_address),
            "entry_point_selector": convert_to_felt(invoke.entry_point_selector),
            "calldata": [convert_to_felt(i) for i in invoke.calldata],
        }
    return {
        "contract_address": convert_to_felt(tx.to_addr),
        "entry_point_selector": convert_to_felt(tx.selector),
        "calldata": [convert_to_felt(i) for i in tx.calldata],
    }
```

This **does not**:
```
    if isinstance(tx, InvokeFunction):
        return {
            "contract_address": convert_to_felt(tx.contract_address),
            "entry_point_selector": convert_to_felt(tx.entry_point_selector),
            "calldata": [convert_to_felt(i) for i in tx.calldata],
        }
    return {
        "contract_address": convert_to_felt(tx.to_addr),
        "entry_point_selector": convert_to_felt(tx.selector),
        "calldata": [convert_to_felt(i) for i in tx.calldata],
    }
```

>  /Users/jakubptak/work/starknet/starknet_python_sdk/starknet_py/net/full_node_client.py:447:52 - error: Cannot access member "contract_address" for type "Call"
    Member "contract_address" is unknown (reportGeneralTypeIssues)
>  /Users/jakubptak/work/starknet/starknet_python_sdk/starknet_py/net/full_node_client.py:448:56 - error: Cannot access member "entry_point_selector" for type "Call"
    Member "entry_point_selector" is unknown (reportGeneralTypeIssues)

Reply:
Well that's weird, but I guess we cannot do anything about it.